lib/exercism/github.rb

```diff
@@ -21,18 +21,23 @@ def self.authenticate(code, client_id, client_secret)
       req.headers['User-Agent'] = user_agent
       req.body = { client_id: client_id, client_secret: client_secret, code: code }.to_json
     end
-    access_token = JSON.parse(response.body)['access_token']
+    JSON.parse(response.body)['access_token']
+  end
 
+  def self.user_info(access_token)
     conn = connect_to 'https://api.github.com'
     response = conn.get do |req|
       req.url '/user'
       req.headers['User-Agent'] = user_agent
       req.params['access_token'] = access_token
     end
-    result = JSON.parse(response.body)
-    [result['id'], result['login'], result['email'], result['avatar_url']]
+    if response.status != 200
```

Comment:
You could also use `!response.success?`, which conveys the intent a little more clearly to people who don't know that `200` means success.

Then you might also reverse the order, putting the positive case first. ([ruby style guide reference](https://github.com/bbatsov/ruby-style-guide#no-else-with-unless))